gosu-core-api/src/main/gosu/gw/lang/enhancements/CoreIterableEnhancement.gsx

```diff
@@ -557,4 +558,25 @@ enhancement CoreIterableEnhancement<T> : java.lang.Iterable<T> {
     }
     return retList
   }
-}
\ No newline at end of file
+
+  /**
+   * takes two lists and returns a list of corresponding <code>gw.util.Pair</code>s.
+   * If one input list is short, excess elements of the longer list are discarded.
+   */
+  function zip<R>( other : Iterable<R>) : Iterable<Pair<T,R>> {
+    if (other == null) {
+      throw new NullPointerException("other should be non-null")
+    }
+
+    var zippedLength = this.Count < other.Count ? this.Count : other.Count
+    var zipped = new ArrayList<Pair<T,R>>(zippedLength)
+
```

Comment:
Just a drive-by comment:

Regardless of ArrayList vs LinkedList, this as implemented should return List rather than Iterable. If it declares returning Iterable, a caller should assume an implementation which would allow for an early break out of a loop. Returning List expresses that your return value will be complete rather than lazily iterated. Note that this is how where() is declared down below.

If it were up to me, I'd declare this as returning Iterable and then write an Iterator-based implementation. That's a lot more code though.